regression/Makefile

```diff
@@ -14,6 +14,7 @@ DIRS = cbmc \
        goto-diff \
        test-script \
        goto-analyzer-taint \
+       goto-cc \
```

Comment:
As discussed out-of-band: please explain this in the commit message and/or rename the directory to clarify that these are GCC/Clang-only tests as they use GCC/Clang-only command-line options.